agent/Dockerfile

```
@@ -134,28 +146,58 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN mkdir -p /criu-install/usr/local/bin \
+    && cp -L "$(command -v tar)" /criu-install/usr/local/bin/tar
+
+# criu's and tar's own shared-library dependencies (libbsd, libgnutls,
+# libnftables, libnl-3, libnet, libuuid, libacl, libselinux, and their
+# transitive deps) must be resolved here, against this stage's older glibc --
+# not later in the agent stage, whose ldd would resolve the same library
+# names against its own (newer) filesystem and silently reintroduce the
+# GLIBC_2.38 floor this stage exists to avoid. The agent stage's closure step
+# is no-clobber, so it fills in gaps (ip, cuda-checkpoint) without
+# overwriting what's collected here.
+RUN set -eu; \
```

Hadolint (2.15.1)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)